src/types/Post.ts

```diff
@@ -0,0 +1,11 @@
+import { Comment } from './Comment';
+import { User } from './User';
+
+export interface Post {
+  userId: number,
+  id: number,
+  title: string,
+  body: string,
+  user: User | null,
+  comments: Comment[] | [],
```

Comment:
you don't need `| []` here
`Comment[]` can be an array with comments or an empty array